### PR TITLE
Add backend architecture section

### PR DESCRIPTION
**Pull request by muItilingualism**
This section will explain the backend architecture and how it functions at a high-level with focus on separation of concerns and system layering.

It may encroach on the "Design decisions" section.

Please review.

### CODE REVIEW

**Review by christofferfink — COMMENTED** [review]
Der kunne godt bruges lidt "små overskrifter"

Det er nærmest allerede gjort fx "Security is an important concern in the system" kunne under-overskriften fx være security - The backbone of our backend is composed  kunne den være "backbone" - The core of the system consists of Services "services" osv

File: report.md [hunk 1]
```diff
@@ -50,6 +51,90 @@
...
+
+A key strength of the architecture is its modular organization. Rather than structuring components by their purpose, they are organized by domain. This approach enhances scalability and maintenance, as each domain — such as authentication or user management — has its own self-contained set of resources, controllers, and services. This structure makes it easier to add new features or modules and simplifies the process of maintaining and updating existing ones.
+
+While the current implementation uses a direct connection to the database through entity classes, the architecture is designed with future improvements in mind. We plan to introduce a repository layer, which will further abstract data access from business logic, enhancing the system's flexibility and maintainability.
+
```
Comment (muItilingualism) [inline report.md:64]: resolve duplicate

**Review by HenrikSkindhoj — COMMENTED** [review]
Line 63 and 65 starts the same way.
Looks good otherwise

**Review by Hamsterpants — APPROVED** [review]
LGTM lala ;-P

**Review by HenrikSkindhoj — APPROVED** [review]
LGTM

**Pull request merged**